Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -40,84 +44,126 @@ open class _ChatChannelListItemView<ExtraData: ExtraDataTypes>: _ChatChannelSwip
+        // Title label
+        constraintsToActivate += [
+            // Bottom of the label is aligned with avatar vertical center
+            titleLabel.lastBaselineAnchor.pin(equalTo: visualCenterGuide.topAnchor),
+
+            // Pin the title label leading anchor to avatar's trailing + spacing
+            titleLabel.leadingAnchor.pin(equalToSystemSpacingAfter: avatarView.trailingAnchor),
+            
+            // Title label top should always be inside the cell
+            titleLabel.topAnchor.pin(greaterThanOrEqualTo: cellContentView.layoutMarginsGuide.topAnchor)
+        ]
+
+        // Subtitle label
+        constraintsToActivate += [
+            // Top of the label is aligned with avatar vertical center
+            subtitleLabel.topAnchor.pin(equalTo: visualCenterGuide.bottomAnchor),
+            
+            // Pin the subtitle label leading anchor to avatar's trailing + spacing
+            subtitleLabel.leadingAnchor.pin(equalToSystemSpacingAfter: avatarView.trailingAnchor)
+        ]
+
+        // Unread count view
+        constraintsToActivate += [
+            // Pin the label to the trailing anchor
+            unreadCountView.trailingAnchor.pin(equalTo: cellContentView.layoutMarginsGuide.trailingAnchor),
+            
+            // Align it vertically with the title
+            unreadCountView.centerYAnchor.pin(equalTo: titleLabel.centerYAnchor),
+            
+            // Title label shouldn't overlap
+            unreadCountView.leadingAnchor.pin(greaterThanOrEqualToSystemSpacingAfter: titleLabel.trailingAnchor)
+        ]
+
+        // Set titleLabel compression resistance smaller than the unread count view
+        titleLabel.setContentCompressionResistancePriority(.streamLow, for: .horizontal)
+        unreadCountView.setContentCompressionResistancePriority(.streamRequire, for: .horizontal)
+
+        // Timestamp label
+        constraintsToActivate += [
+            // Pin the label to the trailing anchor
+            timestampLabel.trailingAnchor.pin(equalTo: cellContentView.layoutMarginsGuide.trailingAnchor),
+            
+            // Align it vertically with the subtitle
+            timestampLabel.centerYAnchor.pin(equalTo: subtitleLabel.centerYAnchor),
         
-        container.centerStackView.isHidden = false
-        container.centerStackView.addArrangedSubview(containerCenterView)
+            // Subtitle label shouldn't overlap
+            timestampLabel.leadingAnchor.pin(greaterThanOrEqualToSystemSpacingAfter: subtitleLabel.trailingAnchor)
+        ]
+
+        // Set subtitleLabel compression resistance smaller than the timestamp label
+        subtitleLabel.setContentCompressionResistancePriority(.streamLow, for: .horizontal)
+        timestampLabel.setContentCompressionResistancePriority(.streamRequire, for: .horizontal)
+
+        NSLayoutConstraint.activate(constraintsToActivate)
     }
     
     override open func updateContent() {
```

Comment:
The "NoChannel" text is strange when the channel is missing. We should show just a black view when the channel is missing. Don't forget to reset all subviews not just the title label, the item view gets reused.

Reply:
@VojtaStavik ✅